Join Tables

anagudelogu opened the pull request:
# In This Milestone I:
## Checked the following tasks:

 - [x] Create a table named vets with the following columns:
   - id: integer (set it as autoincremented PRIMARY KEY)
   - name: string
   - age: integer
   - date_of_graduation: date
![0401](https://user-images.githubusercontent.com/77069011/177774419-9a943812-b833-4c06-8808-116a522e4352.png)
- [x] There is a many-to-many relationship between the tables species and vets: 
   - A vet can specialize in multiple species
   - Species can have multiple vets specialized in it. 
   - Create a "join table" called specializations to handle this relationship.
![Screenshot from 2022-07-07 11-54-33](https://user-images.githubusercontent.com/77069011/177828651-435c1c02-8445-446d-bcee-79a280e44c9a.png)
- [x] There is a many-to-many relationship between the tables animals and vets:
   -  An animal can visit multiple vets.
   - One vet can be visited by multiple animals.
   - Create a "join table" called visits to handle this relationship.
   - It should also keep track of the date of the visit.
![Screenshot from 2022-07-07 11-54-41](https://user-images.githubusercontent.com/77069011/177828725-f45b527c-01c8-4b0b-ad46-95fed070dbb5.png)
- [x] Insert required data into `vets`, `specialization` and `visits` tables.
- [x] Write queries to answer the following:
   - Who was the last animal seen by William Tatcher?
![0404](https://user-images.githubusercontent.com/77069011/177826718-f939cb3d-6924-4bba-bc77-ffa0932bd615.png)
   - How many different animals did Stephanie Mendez see?
![0405](https://user-images.githubusercontent.com/77069011/177826731-22b4b69e-85d4-4f43-8a8b-630db1be05cf.png)
   - List all vets and their specialties, including vets with no specialties.
![0406](https://user-images.githubusercontent.com/77069011/177826780-b9147bbd-92ec-46cb-9903-7b0e2a258e67.png)
   - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
![0407](https://user-images.githubusercontent.com/77069011/177826807-70f2fbde-bd20-4679-814f-d97f6e98d95d.png)
   - What animal has the most visits to vets?
![0408](https://user-images.githubusercontent.com/77069011/177826816-5bee00c0-61cd-4b24-ae4f-af34c0abad26.png)
   - Who was Maisy Smith's first visit?
![0409](https://user-images.githubusercontent.com/77069011/177826903-c6245d15-c6b7-40b4-9a30-6119e77026ad.png)
   - Details for most recent visit: animal information, vet information, and date of visit.
![Screenshot from 2022-07-07 13-17-52](https://user-images.githubusercontent.com/77069011/177842212-b359d5b1-d40d-44b7-ac81-185d6b5e006a.png)
   - How many visits were with a vet that did not specialize in that animal's species?
![Screenshot from 2022-07-07 13-59-41](https://user-images.githubusercontent.com/77069011/177858616-3824c525-c718-4f6b-8616-aed99a920ddc.png)
   - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
![0412](https://user-images.githubusercontent.com/77069011/177859261-8dbaa777-ab43-4902-831d-c260334b1d75.png)

anagudelogu committed:
Create a new table Vets and Join tables for requested relations

anagudelogu committed:
Add required data to vets, specialization and visits tables

anagudelogu committed:
Add queries and their respective screenshots

anagudelogu committed:
Fix PRIMARY KEY in table visits

anagudelogu committed:
Add missing query screenshots

EricMbouwe reviewed (approved):
Hi @anagudelogu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters Ok
- Really good PR
- Added all tables and Queries
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

anagudelogu committed:
Fixed a typo inserting the data

anagudelogu commented:
Hi @EricMbouwe  I just noticed some errors populating the visits table, I missed one entry and misspelled another one, I just fixed that and updated the queries.

anagudelogu committed:
Updated queries affected by typo on data and uploaded new images
